### PR TITLE
[BEAM-2745] add BeamRecordSqlType.getFieldTypeByIndex()

### PR DESCRIPTION
**Pull request by xumingming**
Summary:

* BeamSqlRecordType -> BeamRecordSqlType
* fieldsName -> fieldNames
* fieldsType -> fieldTypes
* add getFieldTypeByIndex()

### CODE REVIEW

**Comment by xumingming** [comment]
R: @takidau @XuMingmin

File: sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java [hunk 1]
```diff
@@ -49,22 +49,22 @@ public void validateValueType(int index, Object fieldValue)
...
     */
    public BeamRecordCoder getRecordCoder(){
-     return BeamRecordCoder.of(this, fieldsCoder);
+     return BeamRecordCoder.of(this, fieldCoders);
    }
 
-   public List<String> getFieldsName(){
-     return fieldsName;
+   public List<String> getFieldNames(){
+     return fieldNames;
```
Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: Wrap in com.google.common.collect.ImmutableList?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java [hunk 2]
```diff
@@ -49,22 +49,22 @@ public void validateValueType(int index, Object fieldValue)
     * Get the coder for {@link BeamRecordCoder}.
     */
    public BeamRecordCoder getRecordCoder(){
-     return BeamRecordCoder.of(this, fieldsCoder);
+     return BeamRecordCoder.of(this, fieldCoders);
    }
 
-   public List<String> getFieldsName(){
-     return fieldsName;
+   public List<String> getFieldNames(){
```
Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: Looking at the other names below, it occurs to me that field names are just being referred to as fields. We should be consistent, and either call this one getFields(), or else rename those getFieldNameByIndex, etc.

Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: field doesn't equal with field name, which happens to be true here. For example, _field_ consists of field name and field type in sql.

Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: But BeamSqlRecordType doesn't override getFieldByIndex to return name + type, it still just returns the name, doesn't it? It sounds like the existing methods that just talk about fields are the ones that should probably be renamed to talk about field names. What do you think?

Comment (xumingming) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: rename to `getFieldNameByIndex` sounds reasonable to me.

Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: +1 for `getFieldNameByIndex` which is clear.

**Comment by takidau** [comment]
Merged, feel free to close. Thank you!

**Pull request closed without merging**